### PR TITLE
Fix interface breaks with Exponential integrators

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
They had a few non-conformities that were causing issues once no recompile mode was enabled.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Fixes https://github.com/SciML/OrdinaryDiffEq.jl/issues/1732

File: src/algorithms.jl [hunk 1]
```diff
@@ -3856,50 +3856,49 @@ RosenbrockW6S4OS(;chunk_size=Val{0}(),autodiff=true, standardtag = Val{true}(),
...
     iop::Int
     autodiff::Bool
     chunksize::Int
   end
   @eval $Alg(;adaptive_krylov=true, m=30, iop=0, autodiff=true, standardtag = Val{true}(), concrete_jac = nothing,
-              chunksize=0, diff_type = Val{:forward}) =
-              $Alg{diff_type,_unwrap_val(standardtag),_unwrap_val(concrete_jac)}(adaptive_krylov, m, iop, _unwrap_val(autodiff), chunksize)
+              chunk_size=Val{0}(), diff_type = Val{:forward}) =
+              $Alg{_unwrap_val(chunk_size),_unwrap_val(autodiff),diff_type,
+              _unwrap_val(standardtag),_unwrap_val(concrete_jac)}(adaptive_krylov, m, iop)
```
Comment (YingboMa) [inline src/algorithms.jl:3894]: Remember to pass `autodiff` and `chunksize` as well here.

Comment (ChrisRackauckas) [reply src/algorithms.jl:3894]: Those should all be removed: all of the other algorithms only have it type parameters 😰

**Comment by ChrisRackauckas** [comment]
I considered this non-breaking because the algorithms were defined in the docs like this already 😅. I guess those are just not options that are used much (they are only used for the exponential Rosenbrock form)

**Pull request merged**